proto/cacheclient.proto

```diff
@@ -402,16 +402,99 @@ message _ListRemoveRequest {
 
 message _ListRemoveResponse {}
 
+// List fetch + retain index behavior
+// This follows redis behavior for the most part, differences
+// 1. end_index is exclusive, unless None
+// 2. `start_index` == 0, then we start at the front of list
+// 2. `end_index` == None, start at the end of the index, inclusive
+/*
+  list_ = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+  # Using the slice notation:
+  # start:end
+
+
+  # start=0, end=None
+  list_[0:] == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+
+  # start=5, end=None
+  list_[5:] == [5, 6, 7, 8, 9]
+
+  # start=5, end=0
+  list_[5:0] == []
+
+  # start=-5, end=None
+  list_[-5:] == [5, 6, 7, 8, 9]
+
+  # start=-5, end=0
+  list_[-5:0] == []
+
+
+  # start=0, end=5
+  list_[0:5] == [0, 1, 2, 3, 4]
+
+  # start=0, end=5
+  list_[0:5] == [0, 1, 2, 3, 4]
```

Comment:
these are copypasted, and if we're going to use inclusive numbers for end is this actually true?